src/files-breadcrumbs.tsx

```diff
@@ -49,19 +49,47 @@ function useHostname() {
 
 // eslint-disable-next-line max-len
 export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: string[], showHidden: boolean, setShowHidden: React.Dispatch<React.SetStateAction<boolean>>}) {
+    const [editMode, setEditMode] = React.useState(false);
+    const [newPath, setNewPath] = React.useState(path.join("/"));
     const hostname = useHostname();
 
     function navigate(n_parts: number) {
         cockpit.location.go("/", { path: encodeURIComponent(path.slice(0, n_parts).join("/")) });
     }
 
+    const handleInputKey = (event: React.KeyboardEvent<HTMLInputElement>) => {
+        if (event.key === "Enter") {
+            changePath();
+        } else if (event.key === "Escape") {
+            cancelPathEdit();
+        }
+    };
+
+    const changePath = () => {
+        setEditMode(false);
+        // HACK: strip trailing / to circumvent the path being `//` in breadcrumbs
+        cockpit.location.go("/", { path: encodeURIComponent(newPath.replace(/\/$/, '')) });
+    };
+
+    const cancelPathEdit = () => {
+        setEditMode(false);
+        setNewPath(path.join("/"));
```

Comment:
Why that? It's not applied anyway, and may do an extra re-render?

Reply:
I want to undo all the changes one does, so say your path is /home/admin, you edit change to /tmp/foo and then cancel then I want to restore the input on the next edit back to the original state.

Reply:
Yes, that's why I meant above that the state should be first-time initialized to `null`, and set to the current path when clicking the edit button. I.e. this state feels too "global"/persistent right now.

Not a veto, it's just a bit brittle that way.